site/content/installation/ngf-images/jwt-token-docker-secret.md

```diff
@@ -0,0 +1,171 @@
+- The certificate (`nginx-repo.crt`) and key (`nginx-repo.key`) for each NGINX Gateway Fabric instance, used to list the available image tags from the Docker registry API.
+
+## Prepare NGINX Gateway Fabric
+
+1. Choose your desired [NGINX Gateway Fabric Image](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/CHANGELOG.md).
+1. Log into the [MyF5 Portal](https://my.f5.com/), navigate to your subscription details, and download the relevant .cert, .key and .JWT files.
+1. Create a Kubernetes secret using the JWT token. You should use `cat` to view the contents of the JWT token and store the output for use in later steps.
+1. Ensure there are no additional characters or extra whitespace that might have been accidentally added. This will break authorization and prevent the NGINX Gateway Fabric image from being downloaded.
+1. Modify your deployment (manifest or Helm) to use the Kubernetes secret created in step 3.
+1. Deploy NGINX Gateway Fabric into your Kubernetes cluster and verify that the installation has been successful.
+
+## Using the JWT token in a Docker Config Secret
+
+1. Create a Kubernetes `docker-registry` secret type on the cluster, using the JWT token as the username and `none` for password (as the password is not used).  The name of the docker server is `private-registry.nginx.com`.
+
+    ```shell
+    kubectl create secret docker-registry regcred --docker-server=private-registry.nginx.com --docker-username=<JWT Token> --docker-password=none [-n nginx-gateway]
+    ```
+
+    It is important that the `--docker-username=<JWT Token>` contains the contents of the token and is not pointing to the token itself. Ensure that when you copy the contents of the JWT token, there are no additional characters or extra whitespaces. This can invalidate the token and cause 401 errors when trying to authenticate to the registry.
+
+1. Confirm the details of the created secret by running:
+
+    ```shell
+    kubectl get secret regcred --output=yaml
+    ```
+
+1. You can now use the newly created Kubernetes secret in Helm and manifest deployments.
+
+---
+
+## Manifest Deployment
+
+The page ["Installation with Manifests"](https://docs.nginx.com/nginx-gateway-fabric/installation/installing-ngf/manifests/) explains how to install NGINX Gateway Fabric using manifests. The following snippet is an example of a deployment:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxGateway
+metadata:
+  name: nginx-gateway-config
+  namespace: nginx-gateway
+  labels:
+    app.kubernetes.io/name: nginx-gateway
+    app.kubernetes.io/instance: nginx-gateway
+    app.kubernetes.io/version: "edge"
+spec:
+  logging:
+    level: info
+```
+
+The `imagePullSecrets` and `containers.image` lines represent the Kubernetes secret, as well as the registry and version of the NGINX Gateway Fabric we are going to deploy.
+
+---
+
+## Helm Deployment
```

Comment:
We already have a helm installation guide, so this section is unnecessary.

Reply:
its specific to plus installation. Again, similar to NIC so I added it

Reply:
We don't have to mirror NIC exactly. It was meant mainly as a reference for the installation steps. I'd rather avoid duplication that just copy/paste without added value.

Reply:
agreed, but it only specifies update to `values.yaml` in terms of plus installation. It would be easier to have good flow of instructions than hopping between pages to reference something.
But again, let me know i'll move this section to helm installation docs.

Reply:
Agreed about flow, but I think we should probably have all installation-related instructions in one doc. This doc can be about acquiring the token and putting it into a secret. I'm okay with the installation doc linking out to this one for creating the secret, and then still having the actual helm install command in the original doc (for this specific installation case).

I'll defer to the team though if they have other preferences.

Reply:
I still think we should remove the installation instructions from this guide. It's duplicated info that we now have to keep updated in multiple places.

This is also specific to helm, but a user could also install using manifests instead. I say we link back to the installation guide now that the Secret has been created.  Still open to hear what other think, though.

Reply:
To weigh in on this, when writing/rewriting documentation, the documentation team often thinks about what the "critical paths" are. Where possible, we try to remove redundancy and only offer relevant information where it's needed.

Redundant and duplicate information becomes a maintenance concern if the instruction ever changes: the same set of instructions then needs to be updated everywhere it was replicated. If it's not, then there's drift. You could create an include file (which we mostly use for reference information), or you have to proactively look for these redundant instructions and create a process for doing so every time a possibly redundant piece of information is updated anywhere in the documentation set.

Unless there are deviations on the critical path necessary to swap or add a component, it's easiest to compartmentalize it on its own document. If it warrants a mention in the "base" document (The general purpose installation guide), then you stick to it being a mention as best you can - "If you would like to take advantage of feature X, you will need to install widget Y" and link to the page for widget Y, which includes the "changed" steps.

Reply:
Makes sense. In this case, N+ requires some sort of extra step in order to pull the image, so this doc isn't necessarily a change from the base install (of N+), but rather an additional setup. Which is why I feel that the installation step can still live in the main doc, and link to this doc to say "how to set things up before installing".

Reply:
Yep, that's my instinct too. I just wanted to explain the context as to why!

Reply:
will update this